src/components/flex/Flex.spec.jsx

```diff
@@ -101,4 +101,75 @@ describe('<Flex>', () => {
         .is('ul')
     ).toEqual(true);
   });
+
+  it('renders responsive props passed as object', () => {
+    const component = shallow(
+      <Flex
+        fullWidth={{
+          sm: true,
+          md: false,
+          lg: true,
+          xl: true,
+        }}
+      >
+        <div>test</div>
+      </Flex>
+    );
+
+    expect(
+      component.findWhere(
+        node =>
+          node.prop('className') ===
+          'sg-flex sg-flex--full-width md:sg-flex--auto-width lg:sg-flex--full-width xl:sg-flex--full-width'
+      )
+    ).toHaveLength(1);
+  });
+
+  it('renders responsive class names with prop passed as array', () => {
+    const component = shallow(
+      <Flex fullWidth={[true, true, false, false]}>
+        <div>test</div>
+      </Flex>
+    );
+
+    expect(
+      component.findWhere(
+        node =>
+          node.prop('className') ===
+          'sg-flex sg-flex--full-width md:sg-flex--full-width lg:sg-flex--auto-width xl:sg-flex--auto-width'
+      )
+    ).toHaveLength(1);
+  });
+
+  it('renders responsive class names with prop passed as array', () => {
+    const component = shallow(
+      <Flex fullWidth={[true, true, false, false]}>
+        <div>test</div>
+      </Flex>
+    );
+
+    expect(
+      component.findWhere(
+        node =>
+          node.prop('className') ===
+          'sg-flex sg-flex--full-width md:sg-flex--full-width lg:sg-flex--auto-width xl:sg-flex--auto-width'
+      )
+    ).toHaveLength(1);
+  });
+
+  it('renders responsive class names when passed null or undefined as item', () => {
```

Comment:
I think that only null was tested here. I would test 3 situations:

- `<Flex fullWidth={[true, true, null, false]}>`
- `<Flex fullWidth={[true, true, undefined, false]}>`
- `<Flex fullWidth={[,,,false]}>` - this is almost the same as above but it might be a good example of use

Reply:
I would also test cases of `<Flex fullWidth={[false,false,true,true]}>` and `<Flex fullWidth={[,,,,]}>`

Reply:
@domidomi @coderitual 
Wouldn't be sufficient to this one case but with 3 approaches?

- `<Flex fullWidth={[true, true, null, false]}>`
- `<Flex fullWidth={[true, true, undefined, false]}>`
- `<Flex fullWidth={[true, true, , false]}>`

I have feeling that other cases can be redundant.

Reply:
```
it('renders responsive class names when passed null or undefined as item, or sparse array', () => {
    const componentWithNull = shallow(
      <Flex fullWidth={[true, true, null, false]}>
        <div>test</div>
      </Flex>
    );

    const componentWithUndefined = shallow(
      <Flex fullWidth={[true, true, undefined, false]}>
        <div>test</div>
      </Flex>
    );

    const componentWithSparseArray = shallow(
      // eslint-disable-next-line no-sparse-arrays
      <Flex fullWidth={[true, true, , false]}>
        <div>test</div>
      </Flex>
    );

    [
      componentWithNull,
      componentWithUndefined,
      componentWithSparseArray,
    ].forEach(component => {
      expect(
        component.findWhere(
          node =>
            node.prop('className') ===
            'sg-flex sg-flex--full-width md:sg-flex--full-width xl:sg-flex--auto-width'
        )
      ).toHaveLength(1);
    });
  });
```

Reply:
I have a feeling that testing different combinations of they ways we pass responsive props is crucial for testing api. not only null vs undefined vs omitting but also different placement of those values in array e.g all 4 placements in array but also what if I pass more than 4 and only 1 value.

Reply:
Ok , I think below list includes all cases for array and object notation. Let me know if we need to cover something more.

Object notation:
- with all breakpoints
- with some breakpoints
- with no breakpoints
- with invalid breakpoints

Array notation:
- with all breakpoints
- with null breakpoints
- with undefined breakpoints
- with skipped breakpoints (sparse array)
- with no breakpoints
- with too many breakpoints

Reply:
cool, how about using `test.each` for all those cases above (I believe we havent used it yet but it might be good use case)?